### PR TITLE
bpo-41288: Fix a crash in unpickling invalid NEWOBJ_EX.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue41288

Automerge-Triggered-By: @tiran

### CODE REVIEW

File: Modules/_pickle.c [hunk 1]
```diff
@@ -5988,23 +5988,30 @@ load_newobj_ex(UnpicklerObject *self)
...
-        Py_DECREF(args);
-        Py_DECREF(cls);
         PyErr_SetString(st->UnpicklingError,
                         "NEWOBJ_EX class argument doesn't have __new__");
-        return -1;
+        goto error;
+    }
+    if (!PyTuple_Check(args)) {
+        PyErr_Format(st->UnpicklingError,
+                     "NEWOBJ_EX args argument must be a type, not %.200s",
```
Comment (tiran) [inline Modules/_pickle.c]:
```suggestion
                     "NEWOBJ_EX args argument must be a tuple, not %.200s",
```

File: Modules/_pickle.c [hunk 2]
```diff
@@ -5988,23 +5988,30 @@ load_newobj_ex(UnpicklerObject *self)
...
+                     Py_TYPE(args)->tp_name);
+        goto error;
+    }
+    if (!PyDict_Check(kwargs)) {
+        PyErr_Format(st->UnpicklingError,
+                     "NEWOBJ_EX kwargs argument must be a type, not %.200s",
```
Comment (tiran) [inline Modules/_pickle.c]:
```suggestion
                     "NEWOBJ_EX kwargs argument must be a dict, not %.200s",
```

**Review by tiran — CHANGES_REQUESTED** [review]

File: Modules/_pickle.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_pickle.c]: Good catch! I wrote this code at late night yesterday. :sleeping:

**Review by serhiy-storchaka — COMMENTED** [review]
I have made the requested changes; please review again.

**Review by tiran — APPROVED** [review]
Excellent, thank you!

**Pull request merged**